src/simplification/Simplify.ts

```diff
@@ -0,0 +1,31 @@
+
+import { CommonModel, Schema } from '../models';
+import simplifyTypes from './SimplifyTypes';
+let anonymCounter = 1;
+
+/**
+ * Simplifies a schema into instances of CommonModel.
+ * 
+ * @param schema to simplify
+ */
+export function simplify(schema : Schema | boolean) : CommonModel[] {
+  let models : CommonModel[] = [];
+  let model = new CommonModel();
+  model.originalSchema = Schema.toSchema(schema);
+  model.type = simplifyTypes(schema);
+  if(typeof schema !== "boolean"){
+    //All schemas of type object MUST have ids, for now lets make it simple
+    if(model.type !== undefined && model.type.includes("object")){
+      let schemaId = schema.$id ? schema.$id : `anonymSchema${anonymCounter++}`;
+      if(typeof schema !== "boolean"){
+        schema.$id = schemaId;
+      }
+      model.$id = schemaId;
+    } else if (schema.$id !== undefined){
+      model.$id = schema.$id;
+    }
+  }
+
+  models.push(model);
```

Comment:
For what we have array there? As I see you only perform action on one schema with one result model.

Reply:
Gonna be introduced in the upcoming PR's, but with properties etc it needs to be able to create new models on the fly 😄